gms/endpoint_state.hh

```diff
@@ -119,20 +119,18 @@ public:
 
 public:
     std::string_view get_status() const noexcept {
-        constexpr std::string_view empty = "";
-        auto* app_state = get_application_state_ptr(application_state::STATUS);
+        constexpr std::string_view empty;
```

Comment:
But default-constructed string_view is not equivalent to empty string, IIUC.

Reply:
That is a good question ... the reason of this change was the clang-tidy reporting "redundant string initialization" if the empty qoutes were used:

> Redundant string initialization (fix available)clang-tidy[readability-redundant-string-init](https://clang.llvm.org/extra/clang-tidy/checks/readability/redundant-string-init.html)

I'm wondering if that does actually make any difference - the empty quoted initializer will have a "valid" data (also null terminated) and size of 0, the default constructor (I believe) will have the data set to null and also length of 0, so the question is whether that makes any difference for the `std::string_view` usage (as the `string_view` is never guaranteed to be null terminated and expecting so would be a bug - and if the length is checked first, both would be "empty" from that point of view).